fastly/resource_fastly_tls_subscription.go

```diff
@@ -68,11 +68,29 @@ func resourceFastlyTLSSubscription() *schema.Resource {
 				Description: "The current state of the subscription. The list of possible states are: `pending`, `processing`, `issued`, and `renewing`.",
 				Computed:    true,
 			},
-			"managed_dns_challenge": {
-				Type:        schema.TypeMap,
-				Description: "The details required to configure DNS to respond to ACME DNS challenge in order to verify domain ownership.",
+			"managed_dns_challenges": {
```

Comment:
As you mentioned, this schema change is a breaking change - to assist in the transition to the new version, it might be good to implement a [`StateUpgrader` function](https://www.terraform.io/docs/extend/resources/state-migration.html) here. The users will still have to update their config files to use the new schema, but at least the statefile can be upgraded automatically

Reply:
But `$ terraform refresh` also updates the state file, right? And any subsequent apply after the version update would override the correct the state file, is what I was seeing in my tests earlier.

Reply:
Fair enough if you've tested it! I didn't know if it would correctly overwrite the old state with the new schema, or if you'd have to migrate it. I can see that would be possible in principle as it's a fully computed value, so we wouldn't need to try and preserve their config file, so happy to leave it if it works 👍🏼